python/kvikio/nvcomp_codec.py

```diff
@@ -196,37 +176,37 @@ def decode_batch(
         if is_host_buffer:
             bufs = [cp.asarray(ensure_contiguous_ndarray_like(b)) for b in bufs]
 
-        # Get uncompressed chunk sizes from the header.
-        uncomp_chunk_sizes = [
-            int(b[: self.HEADER_SIZE_BYTES].view(dtype=cp.uint32)[0]) for b in bufs
-        ]
-        max_chunk_size = max(uncomp_chunk_sizes)
+        # Prepare compressed chunks buffers.
+        comp_chunks = cp.array([b.data.ptr for b in bufs], dtype=cp.uint64)
+        comp_chunk_sizes = cp.array([b.size for b in bufs], dtype=cp.uint64)
```

Comment:
Technically these should probably be [`cp.uintp`](https://numpy.org/devdocs/reference/arrays.scalars.html#numpy.uintp), et seq. throughout:
```suggestion
        comp_chunks = cp.array([b.data.ptr for b in bufs], dtype=cp.uintp)
        comp_chunk_sizes = cp.array([b.size for b in bufs], dtype=cp.uintp)
```

Reply:
Agree about pointers, updated the code accordingly. Would still prefer using `uint64` type for sizes, as those are actual sizes, not pointers.